api-report/merge-tree.api.md

```diff
@@ -180,6 +182,8 @@ export class Client extends TypedEventEmitter<IClientEvents> {
     serializeGCData(handle: IFluidHandle, handleCollectingSerializer: IFluidSerializer): void;
     // (undocumented)
     readonly specToSegment: (spec: IJSONSegment) => ISegment;
+    // @internal
+    get startOfTreeSegment(): ISegment;
```

Comment:
It looks like this is only used in sequence to specify a segment+offset. I think I'd rather push the notion of a 'segoff spec' down into merge-tree, and make appropriate local reference position creation functions allow 'start' / 'end'. It seems plausible application code might want to refer to these refs at some point in the future if we make the concept more mature.

Reply:
this won't prevent application code from getting their hands on a start or end segment (they can create a local ref to it, then ask for the seg+offset of that ref), so I think we need to add some docs on the semantics of these endpoint segments. The fields added to `ISegment` seem like a reasonable place for that.

Reply:
I think this strategy can also fix the kludging around invalid access of client/mergeTree in the `getSlideTo` functions. I think all usages of them boil down to wanting to create a new local reference position (at potentially the start/end of the string). If we make them return start or end accordingly, things should work out more nicely.

Reply:
Implemented my interpretation of this in https://github.com/microsoft/FluidFramework/pull/16310/commits/147deb735841557cfe83a2a9cce14c6432e35835. Though I'm unsure if you mean for there to be a way to construct an endpoint reference in e.g. `createLocalReferencePosition`